token/transfer-hook/cli/src/meta.rs

```diff
@@ -0,0 +1,318 @@
+use {
+    serde::{Deserialize, Serialize},
+    solana_sdk::pubkey::Pubkey,
+    spl_tlv_account_resolution::{account::ExtraAccountMeta, seeds::Seed},
+    std::{fmt, path::Path, str::FromStr},
+    strum_macros::{EnumString, IntoStaticStr},
+};
+
+#[derive(Debug, Clone, Copy, PartialEq, EnumString, IntoStaticStr, Serialize, Deserialize)]
+#[serde(rename_all = "kebab-case")]
+#[strum(serialize_all = "kebab-case")]
+enum AccountMetaRole {
+    Readonly,
+    Writable,
+    ReadonlySigner,
+    WritableSigner,
+}
+impl AccountMetaRole {
+    fn bools(&self) -> (/* is_signer */ bool, /* is_writable */ bool) {
+        match self {
+            AccountMetaRole::Readonly => (false, false),
+            AccountMetaRole::Writable => (false, true),
+            AccountMetaRole::ReadonlySigner => (true, false),
+            AccountMetaRole::WritableSigner => (true, true),
+        }
+    }
+}
+impl fmt::Display for AccountMetaRole {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
+
+#[derive(Debug, PartialEq, Serialize, Deserialize)]
+struct ExtraAccountMetaConfigFile {
+    #[serde(alias = "extraMetas")]
+    extra_metas: Vec<ExtraAccountMetaConfig>,
+}
+
+#[derive(Debug, PartialEq, Serialize, Deserialize)]
+struct ExtraAccountMetaConfig {
+    pubkey: Option<String>,
+    seeds: Option<Vec<Seed>>,
```

Comment:
nit: how about an enum here since you can't have both? It looks like serde makes this nifty https://serde.rs/enum-representations.html, and it'll make `discriminator_and_address` neater